### PR TITLE
COS-116 Changed functionality of 'mutateIn' to accept single value in list

### PR DESCRIPTION
**Pull request by tigrankhojoyan**

### CODE REVIEW

File: liquibase-couchbase/src/main/java/liquibase/ext/couchbase/validator/MutateInInsertUpsertUniqueValidator.java [hunk 1]
```diff
@@ -15,10 +15,9 @@ public MutateInInsertUpsertUniqueValidator(MutateInType mutateInType) {
     }
 
     @Override
-    public void validate(String path, Value value, List<Value> values) {
+    public void validate(String path, List<Value> values) {
         validatePathPresence(path);
-        validateValuePresence(value);
-        validateNoMultipleValues(values);
+        validateMultipleValuesPresence(values);
```
Comment (DaimonCool) [inline liquibase-couchbase/src/main/java/liquibase/ext/couchbase/validator/MutateInInsertUpsertUniqueValidator.java:20]: Don't we break the logic here? we need to ensure that if we need to insert or upsert via mutate in, we can pass only 1 value, here you delete ` validateNoMultipleValues(values);` it means that if we pass multiple values, there will no be an error. 

We should validate that only 1 value is passed (that list contains only 1 value)


You need to create method like validateOneValue where you will check that list contains only 1 value

File: liquibase-couchbase/src/main/java/liquibase/ext/couchbase/validator/MutateInLongValueValidator.java [hunk 2]
```diff
@@ -17,13 +15,9 @@ public MutateInLongValueValidator(MutateInType mutateInType) {
...
     @Override
-    public void validate(String path, Value value, List<Value> values) {
+    public void validate(String path, List<Value> values) {
         validatePathPresence(path);
-        validateValuePresence(value);
-        validateNoMultipleValues(values);
-        if (value.getType() != DataType.LONG) {
-            throw new MutateInDataTypeNotAllowedException(value.getType(), mutateInType);
-        }
+        validateMultipleValuesPresence(values);
```
Comment (DaimonCool) [inline liquibase-couchbase/src/main/java/liquibase/ext/couchbase/validator/MutateInLongValueValidator.java:20]: Again like in the comment above I think we break logic here. We need to ensure that only 1 value is passed and that it's LONG type. Here you just remove validation on LONG, we need it


You need to create method like validateOneValue where you will check that list contains only 1 value

And check that this value is LONG type (not in the new method above)

File: liquibase-couchbase/src/main/java/liquibase/ext/couchbase/validator/MutateInReplaceValidator.java [hunk 3]
```diff
@@ -15,9 +15,8 @@ public MutateInReplaceValidator(MutateInType mutateInType) {
     }
 
     @Override
-    public void validate(String path, Value value, List<Value> values) {
-        validateValuePresence(value);
-        validateNoMultipleValues(values);
```
Comment (DaimonCool) [inline liquibase-couchbase/src/main/java/liquibase/ext/couchbase/validator/MutateInReplaceValidator.java:20]: Again we need to ensure that only 1 values is passed. 

You need to create method like validateOneValue where you will check that list contains only 1 value

File: liquibase-couchbase/src/main/java/liquibase/ext/couchbase/validator/MutateInValidator.java [hunk 4]
```diff
@@ -42,8 +42,8 @@ protected boolean isValueEmpty(Value value) {
             .isPresent();
     }
 
-    protected void validateNoMultipleValues(List<Value> values) {
-        if (!values.isEmpty()) {
+    protected void validateMultipleValuesPresence(List<Value> values) {
+        if (values.isEmpty()) {
             throw new MutateInMultipleValuesNotAllowedException(mutateInType);
         }
```
Comment (DaimonCool) [inline liquibase-couchbase/src/main/java/liquibase/ext/couchbase/validator/MutateInValidator.java:48]: If values is empty we throw `MutateInMultipleValuesNotAllowedException` ? Please look at the comments above where I explained why it's wrong

File: liquibase-couchbase/src/test/java/system/precondition/SqlCheckPreconditionSystemTest.java [hunk 5]
```diff
@@ -9,28 +9,34 @@
...
 
     @BeforeAll
+    @SneakyThrows
     static void setUp() {
+        bucketOperator.createCollection(COLLECTION_NAME, TEST_SCOPE);
+        TimeUnit.SECONDS.sleep(3L);
+        collection = bucketOperator.getCollection(COLLECTION_NAME, TEST_SCOPE);
         collection.queryIndexes().createPrimaryIndex();
-        clusterOperator.removeAllDocuments(TEST_KEYSPACE);
+        TimeUnit.SECONDS.sleep(3L);
```
Comment (DaimonCool) [inline liquibase-couchbase/src/test/java/system/precondition/SqlCheckPreconditionSystemTest.java:33]: I think the tests should pass without this sleep. Please check

File: liquibase-couchbase/src/main/java/liquibase/ext/couchbase/validator/MutateInInsertUpsertUniqueValidator.java (hunk 1 above)
Comment (tigrankhojoyan) [reply liquibase-couchbase/src/main/java/liquibase/ext/couchbase/validator/MutateInInsertUpsertUniqueValidator.java]: Corrected, thanks!

**Review by CountRick — APPROVED** [review]

**Review by DaimonCool — APPROVED** [review]

**Pull request merged**